lib/Tooling/Refactor/ExtractRepeatedExpressionIntoVariable.cpp

```diff
@@ -265,18 +267,28 @@ ExtractRepeatedExpressionIntoVariableOperation::perform(
   PP.SuppressLifetimeQualifiers = true;
   PP.SuppressUnwrittenScope = true;
   T.print(OS, PP, /*PlaceHolder*/ Name);
+  // FIXME: We should hook into the TypePrinter when moving over to llvm.org
+  // instead and get the offset from it.
+  unsigned NameOffset = StringRef(OS.str()).find(Name);
   OS << " = ";
   E->printPretty(OS, /*Helper=*/nullptr, Context.getPrintingPolicy());
   OS << ";\n";
-  Replacements.emplace_back(SourceRange(InsertionLoc, InsertionLoc), OS.str());
+  Replacements.push_back(RefactoringReplacement(
+      SourceRange(InsertionLoc, InsertionLoc), OS.str(), CreatedSymbol.get(),
+      RefactoringReplacement::AssociatedSymbolLocation(
+          llvm::makeArrayRef(NameOffset), /*IsDeclaration=*/true)));
 
   // Replace the duplicates with a reference to the variable.
-  for (const Expr *E : DuplicateExpressions)
-    Replacements.emplace_back(
+  for (const Expr *E : DuplicateExpressions) {
+    Replacements.push_back(RefactoringReplacement(
         SourceRange(SM.getSpellingLoc(E->getLocStart()),
                     getPreciseTokenLocEnd(SM.getSpellingLoc(E->getLocEnd()), SM,
                                           Context.getLangOpts())),
-        Name);
+        Name, CreatedSymbol.get(),
+        /*NameOffset=*/llvm::makeArrayRef(unsigned(0))));
+  }
 
-  return std::move(Replacements);
+  RefactoringResult Result(std::move(Replacements));
+  Result.AssociatedSymbols.push_back(std::move(CreatedSymbol));
```

Comment:
Minor, but: I think it's worth tying the lifetime of CreatedSymbol to the result as early as possible. Can you directly assign CreatedSymbol into an empty RefactoringResult?